src/lib.rs

```diff
@@ -855,6 +867,17 @@ fn cstr_to_string(ptr: *const libc::c_char) -> Result<Option<String>, Error> {
     Ok(string)
 }
 
+#[cfg(target_os = "windows")]
+#[inline]
+fn wstr_to_string(ptr: *const libc::c_char) -> Result<Option<String>, Error> {
+    let string = if ptr.is_null() {
+        None
+    } else {
+        Some(unsafe { WideCString::from_ptr_str(ptr as _) }.to_string().unwrap())
```

Comment:
Thoughts on this `unwrap()`.

If we replaced `.to_string().unwrap()` with `.to_string()?` it won't compile because `Error` type are not compatible between caller and callee. We could modify our `Error` type, but I feel it like an overkill.

My suggestion is to simply replace `.to_string().unwrap()` with `.to_string_lossy()` [1] which always succeeds. What do you think about that?

[1] https://docs.rs/widestring/0.4.3/widestring/struct.UCString.html#method.to_string_lossy

Reply:
pcap Error contains a MalformedError that represent Utf8 error why not use it ?

Reply:
It might be a good idea, but it's not a UTF8 string we are processing.
The best way here is probably: 
1) Add an error variant that wraps `FromUtf16Error`.
2) Return the above variant when error happens.